### PR TITLE
User-specified mouse behaviour in 3D navigation

### PR DESCRIPTION
**Pull request by harrybraviner**
Allows the user to specify what 3D navigation action they want when left/middle/right click-and-drag-ing. A new preferences menu is added to achieved this. Defaults of OpenSCAD, Blender, and Fusion settings are available, and the user can customize behaviour.

Note that the 'swap mouse buttons' checkbox has been removed because it is now redundant - the user should set this behaviour in the "Mouse" preferences page.

This PR addresses #5709

### CODE REVIEW

File: src/core/Settings.h [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include "gui/input/MouseConfig.h"
```
Comment (t-paul) [inline src/core/Settings.h:3]: The include looks unused, I think we can drop that here (also getting rid of the core->gui dependency).

File: src/gui/QGLView.cc [hunk 2]
```diff
@@ -314,43 +314,77 @@ void QGLView::mouseMoveEvent(QMouseEvent *event)
...
   if (mouse_drag_active) {
     mouse_drag_moved = true;
-    auto button_compare = this->mouseSwapButtons?Qt::RightButton : Qt::LeftButton;
-    if (event->buttons() & button_compare
-#ifdef Q_OS_MACOS
-        && !(event->modifiers() & Qt::MetaModifier)
-#endif
-        ) {
-      // Left button rotates in xz, Shift-left rotates in xy
-      // On Mac, Ctrl-Left is handled as right button on other platforms
```
Comment (t-paul) [inline src/gui/QGLView.cc:324]: What's the replacement for that macOS special case? While we do have full configuration now, it would not be nice forcing all macOS users to use a custom setting for keeping the old behavior. Does this maybe need a special case when defining the `presetSettings`?

**Review by t-paul — CHANGES_REQUESTED** [review]
Looks great! There are just a couple of minor points to go through. My local build just finished, so I'll give it some testing time now...

File: src/core/Settings.cc [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #include "core/Settings.h"
 
+#include "gui/input/MouseConfig.h"
```
[the remark this answers is not in the document]
Comment (harrybraviner) [reply src/core/Settings.cc]: I've moved it. Thanks for pointing this out - I didn't actually know there was a non-GUI mode!

File: src/gui/QGLView.cc (hunk 2 above)
Comment (harrybraviner) [reply src/gui/QGLView.cc:324]: This was a little trickier to resolve. I think the right call is to add the option to define Ctrl+Shift+{Left, Middle, Right} click behaviour. I've now done so. (Additionally, Ctrl+Shift+Middle click is used for zoom in Fusion, so this is necessary to properly mimic that behaviour.)

I considered having the OpenSCAD preset modified between Mac OS and other OSs at compile time, but decided against it. Instead I've added this as a different preset, which is selected by default on a Mac. This has the benefit that a non-Mac user can also select this behaviour (e.g. maybe some Windows/Linux users out there have an injury that makes it harder to operate other buttons on their mouse, and would find this useful).

They key thing is: Mac users should now experience no sudden change of behaviour after this update.

**Comment by harrybraviner** [comment]
@t-paul Ok, I think I've addressed your comments. Let me know if there's anything else that needs fixing. Thanks!

**Review by t-paul — APPROVED** [review]
Looks very nice! I did some comparison of the new default with an older build and that also seems to be good as far as my tests go on Linux. I'd say we go ahead and see what other platforms say. I assume we'll hear about it quite soon in case there are some hidden corner cases but with the configuration nicely listed in the config tables, it should be relatively easy to handle if something comes up.
